Coding/fw/lib/WebServer/Pages.cpp

```diff
@@ -0,0 +1,112 @@
+#include <SPIFFS.h>
+#include "Settings.h"
+
+/* C-Interface ************************************************************************************/
+extern "C"
+{
+}
+
+/* CONSTANTS **************************************************************************************/
+
+/* MACROS *****************************************************************************************/
+
+/* TYPES ******************************************************************************************/
+
+/* PROTOTYPES *************************************************************************************/
+static String pageProcessor(const String &var);
+static void indexPage(AsyncWebServerRequest *request);
+static void errorPage(AsyncWebServerRequest *request);
+
+/* VARIABLES **************************************************************************************/
+
+/* PUBLIC METHODES ********************************************************************************/
+/**************************************************************************************************/
+void Pages::init(AsyncWebServer &webServer)
+{
+    webServer.on("/", HTTP_GET, indexPage);
+
+    webServer.serveStatic("/css/w3.css", SPIFFS, "/css/w3.css", "max-age = 3600");
+    webServer.serveStatic("/pictures/NewTec_Logo.png", SPIFFS, "/pictures/NewTec_Logo.png", "max-age = 3600");
+
+    webServer.onNotFound(errorPage);
+}
+
+/* PROTECTED METHODES *****************************************************************************/
+
+/* PRIVATE METHODES *******************************************************************************/
+
+/* EXTERNAL FUNCTIONS *****************************************************************************/
+
+/* INTERNAL FUNCTIONS *****************************************************************************/
+static String pageProcessor(const String &var)
+{
+    String temp;
+
+    if (var == "STATION_SSID")
+    {
+        temp = WebConfig::getSTA_SSID();
+    }
+    else if (var == "AP_SSID")
+    {
+       temp = WebConfig::getAP_SSID();
+    }
+    else if (var == "AP_PASS")
+    {
+        temp = WebConfig::getAP_PASS();
+    }
+    else if (var == "SERVER_USER")
+    {
+        temp = WebConfig::getWEB_USER();
+    }
+    else if (var == "SERVER_PASS")
+    {
+        temp = WebConfig::getWEB_PASS();
+    }
+    return temp;
+}
+
+static void indexPage(AsyncWebServerRequest *request)
+{
+    if (nullptr == request)
+    {
+        return;
+    }
+
+    if (!request->authenticate(WebConfig::getWEB_USER().c_str(), WebConfig::getWEB_PASS().c_str()))
+    {
+        return request->requestAuthentication();
+    }
+
+    return request->send(SPIFFS, "/index.html", String(), false, pageProcessor);
+}
+
+static void errorPage(AsyncWebServerRequest *request)
+{
+    if (nullptr == request)
+    {
+        return;
+    }
+
+    if (!request->authenticate(WebConfig::getWEB_USER().c_str(), WebConfig::getWEB_PASS().c_str()))
+    {
+        return request->requestAuthentication();
```

Comment:
value return in void

Reply:
https://github.com/me-no-dev/ESPAsyncWebServer#setting-up-the-server

Reply:
Linked example is bad. Note, the requestAuthentication() method returns nothing! ;-)